fullstack_template/server/server.py

```diff
@@ -15,14 +18,29 @@ def home():
 def hello():
    return "Hello World!"
 
+def authenticate(username, password):
+    user = User.query.filter_by(email=username).first()
+    if user and check_password(user.password, password):
+        return user
+
+def identity(payload):
+    user_id = payload['identity']
+    return User.query.get(int(user_id)).email
+
+jwt = JWT(app, authenticate, identity)
+
 @app.route('/register',methods=['POST'])
 def register():
     if request.method == 'POST':
         request_data = request.get_json()
         data = User(first_name=request_data["first_name"],
                     last_name=request_data["last_name"], 
                     email=request_data["email"],
-                    password=hash_password(request_data["password"]))
+                    password=hash_password(request_data["password"]),
+                    major=None)
+        user = User.query.filter_by(email=data.email).first()
+        if user is not None: 
+            return "unsucessful register"
```

Comment:
Can we send a json error message from this?  Can we send a response of 400 and specify that an email already exists?

Reply:
Same applies to the else case

Reply:
Yes, that is a good idea. I will make the changes.